src/ceph-disk/ceph_disk/main.py

```diff
@@ -2545,6 +2567,60 @@ def get_dmcrypt_type(args):
             return None
 
 
+class Secrets(object):
+
+    def __init__(self):
+        secret, stderr, ret = command(['ceph-authtool', '--gen-print-key'])
+        LOG.debug("stderr " + stderr)
+        assert ret == 0
+        self.keys = {
+            'cephx_secret': secret.strip(),
+        }
+
+    def write_osd_keyring(self, keyring, osd_id):
+        command_check_call(
+            [
+                'ceph-authtool', keyring,
+                '--create-keyring',
+                '--name', 'osd.' + str(osd_id),
+                '--add-key', self.keys['cephx_secret'],
+            ])
+        path_set_context(keyring)
+
+    def get_json(self):
+        return bytearray(json.dumps(self.keys), 'ascii')
+
+
+class LockboxSecrets(Secrets):
+
+    def __init__(self, args):
+        super(LockboxSecrets, self).__init__()
+
+        key_size = CryptHelpers.get_dmcrypt_keysize(args)
+        key = open('/dev/urandom', 'rb').read(key_size / 8)
```

Comment:
Sure, at some point the file will be closed when the program exits, in the meantime Python will keep those file handles around. `ceph-disk` has a few places where it just opens files without closing and others where it does the best practice: uses a `with` statement or plainly calls `close()`.

Given that on pr #15585 we are looking at waiting for up to 3 hours, I would be concerned of adding more of these instead of just closing the file.

I also asked about just using Python's random module instead, that would not need to open files like this implementation